Data/Vector.hs

```diff
@@ -1063,6 +1064,18 @@ forM_ :: Monad m => Vector a -> (a -> m b) -> m ()
 {-# INLINE forM_ #-}
 forM_ = G.forM_
 
+-- | /O(n)/ Apply the monadic action to all elements and their indexes of the vector, yielding a
```

Comment:
Other haddock comments and API (`findIndices`) in `vector` spell "indices" instead of "indexes" and AFAIK this is a preferred form in mathematical/technical context. If there is no particular reason to deviate, I suggest to stick to "indices".

Reply:
Also, "their indexes of the vector" sounds a bit strange; "to all elements of the vector and their indices" is better.

Reply:
Fixed. I like "indices" better